apps/condo/domains/common/hooks/useValidations.ts

```diff
@@ -0,0 +1,64 @@
+import { useIntl } from '@core/next/intl'
+import { Rule } from 'rc-field-form/lib/interface'
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+
+type CombinerParams = Array<Rule | Array<Rule>>
+
+type ValidatorHookReturnType = {
+    combiner: (...rules: CombinerParams) => Rule[]
+    messageChanger: (rule: Rule, message: string) => Rule
+    requiredValidator: Rule
+    phoneValidator: Rule
+    emailValidator: Rule
+    trimValidator: Rule
+}
+
+const combiner = (...rules: CombinerParams) => {
+    return rules.map(rulesPart => Array.isArray(rulesPart) ? rulesPart : [rulesPart]).flat()
+}
+
+const messageChanger = (rule: Rule, message: string) => {
+    return { ...rule, message }
+}
+
+export const useValidations: () => ValidatorHookReturnType = () => {
+    const intl = useIntl()
+    const ThisFieldIsRequiredMessage = intl.formatMessage({ id: 'FieldIsRequired' })
+    const PhoneIsNotValidMessage = intl.formatMessage({ id: 'pages.auth.PhoneIsNotValid' })
+    const EmailErrorMessage = intl.formatMessage({ id: 'pages.auth.EmailIsNotValid' })
+
+    const requiredValidator: Rule = {
+        required: true,
+        message: ThisFieldIsRequiredMessage,
+    }
+
+    const phoneValidator: Rule = {
+        validator: (_, value) => {
+            if (!value) return Promise.resolve()
+            const v = normalizePhone(value)
+            if (!v) return Promise.reject(PhoneIsNotValidMessage)
+            return Promise.resolve()
+        },
+    }
+
+    const emailValidator: Rule = {
+        type: 'email',
+        message: EmailErrorMessage,
+    }
+
+    const trimValidator: Rule = {
+        validator: (_, value) => {
+            if (!value || value.trim().length === 0) return Promise.reject(ThisFieldIsRequiredMessage)
+            return Promise.resolve()
+        },
```

Comment:
Maybe it is worth to make function `validator` async and return pure value, so you wouldn't wrap all in `Promise.resolve()` and `Promise.reject()`
E.g.:
```ts
    const phoneValidator: Rule = {
        validator: async (_, value) => {
            if (!value) return
            const v = normalizePhone(value)
            if (!v) throw new Error(PhoneIsNotValidMessage)
        },
    }
```

Reply:
We need to return rejected Promise from validators. We will include promise in any case.